### PR TITLE
feat: txn command to simulate

### PR DESCRIPTION
**Pull request by willemneal**
Part of https://github.com/stellar/soroban-cli/issues/1265
Adds 

- `txn simulate` - simulate and assemble transaction

Future PR:
- `signer::Stellar` trait and a default `InMemory` implementation needed for `txn sign`
- `txn sign` - sign a given transaction
- `txn inspect`
- `txn send` - sends a signed transaction to the PRC (not useful without sign)

Left
- [x] Tests

### CODE REVIEW

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/mod.rs:13]: It's a bit odd that `tx inspect` works like this: `tx inspect decode --type TransactionEnvelope`.

I think we should replace it with `tx decode` and `tx encode` where the type parameter is already filled in.

See this issue that details the commands:
- https://github.com/stellar/soroban-cli/issues/1180

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/mod.rs:15]: The doc comments on these two commands look like a copy-paste from somewhere else and don't make much sense in this context.

File: cmd/soroban-cli/src/commands/tx/send.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/send.rs:25]: If we're going to output the response we should probably output it in JSON rather than the rust debug format, so it is at least usable.

I'm also wondering what's the right thing to return from here. The response makes sense I guess, it actually contains a lot of data that could be useful when scripting.

File: cmd/soroban-cli/src/commands/tx/sign.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/sign.rs:36]: We should leave out the yes, and require the user to specifier the list of signers who should sign, using the same format as expressed here:
- https://github.com/stellar/soroban-cli/issues/1264

And then address interactivity at a later time holistically.

For starters in this PR we could allow just one signer to begin with, and defer to #1264 adding support for multiple.

File: cmd/soroban-cli/src/commands/tx/sign.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/sign.rs:46]: What is the `.trim()` needed for?

File: cmd/soroban-cli/src/commands/tx/xdr.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/xdr.rs:38]: Can we replace these xdr inputs with reading from stdin only, for consistency with the xdr commands? Would rather keep the interface smaller, and then look at improving the interface for passing XDR around separately and holistically across all the XDR commands to make sure what we do works everywhere and is consistent. The API for parameter passing is likely to be different to this to be consistent with the how the XDR subcommand does format selection.

File: cmd/soroban-cli/src/signer.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/signer.rs:84]: The sign function assumes the tx hasn't been wrapped in a fee bump transaction, but ideally it would support all Stellar txs.

**Review by leighmcculloch — COMMENTED** [review]
I pushed some commits getting main merged in.

This PR looks like it's still a little ways away from being ready. Left some inline comments. 

Would be a good idea to add tests for these commands too, from the outside-in.

**Comment by leighmcculloch** [comment]
Given the sensitivity of the sign command, I'd like to shift that work into a separate PR, so that we can get the other commands merged sooner, then follow up and take our time with the sign command. cc @janewang

File: cmd/soroban-cli/src/commands/tx/xdr.rs
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/tx/xdr.rs:61]: It should always be TransactionEnvelope's that we pass around, not Transactions on their own.

File: cmd/soroban-cli/src/commands/tx/sign.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/sign.rs:46]: Had issues with piping to a send command. But perhaps it should be on the input.

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/mod.rs:13]: `decode::Cmd` is not [public](https://github.com/stellar/rs-stellar-xdr/blob/efe941d4f0dc55d555273ff7da2b46626f7f985b/src/cli/mod.rs#L1) so I went with root, which is as you point out less than optimal.

File: cmd/soroban-cli/src/commands/tx/xdr.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/xdr.rs:38]: That's okay with me. `type` and `cat` can work for now. And then when we can properly build a txn envolope it will be the best way to start the piping.

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (leighmcculloch) [reply cmd/soroban-cli/src/commands/tx/mod.rs:13]: Making it public here:
- https://github.com/stellar/rs-stellar-xdr/pull/369

We could remove the inspect/encode/decode from this PR and revisit it once that is merged.

**Comment by willemneal** [comment]
Friend bot seems to be failing.

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (chadoh) [inline cmd/crates/soroban-test/tests/it/integration/tx.rs:14]: Should we remove this?

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (chadoh) [inline cmd/crates/soroban-test/tests/it/integration/tx.rs:29]: And this?

**Review by chadoh — COMMENTED** [review]
Nice clean code! I really like the tests. I have some questions, before I approve.

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (chadoh) [reply cmd/crates/soroban-test/tests/it/integration/tx.rs:38]: Having looked at the code itself, I think this is more a question about where the logic in `run` happens. 

```rs
TransactionEnvelope::Tx(TransactionV1Envelope {
    tx,
    signatures: VecM::default()
})
.to_xdr_base64(xdr::Limits::none())?
```

Why do this in `run`, but not in `run_against_rpc_server`? Why not in `simulate`? 

You might have a great reason for this, but coming in fresh, all of these functions seem like they should do the same thing.

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (willemneal) [reply cmd/crates/soroban-test/tests/it/integration/tx.rs:14]: It's only printed if it fails, but it was just for debugging.

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/mod.rs:6]: It was pending a release of rs-stellar-xdr. For the sake of closing this I say we do it in a follow up PR. I'm pretty sure it's released now.

File: cmd/soroban-cli/src/commands/tx/simulate.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/simulate.rs:46]: Very true!

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (chadoh) [reply cmd/soroban-cli/src/commands/tx/mod.rs:6]: oh, you mean the file itself? I just saw that we have a `tx/inspect.rs` that's not included in the module. But yeah, up to you!

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (willemneal) [reply cmd/crates/soroban-test/tests/it/integration/tx.rs:19]: I should! But it currently expects stdin. I'll refactor so that you can pass it instead of assuming stdin.

File: cmd/crates/soroban-test/tests/it/integration/tx.rs
Comment (willemneal) [reply cmd/crates/soroban-test/tests/it/integration/tx.rs:38]: My thought was that as a library after a simulation you probably wanted an assembled transaction. However, as the output from the CLI you want a transaction envelope to keep the expected type consistent across all `tx` commands.

File: cmd/soroban-cli/src/commands/tx/mod.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/mod.rs:6]: Yeah that should be removed.

File: cmd/soroban-cli/src/commands/tx/simulate.rs
Comment (willemneal) [reply cmd/soroban-cli/src/commands/tx/simulate.rs]: removed it! Was a bit silly since it was just one line.

**Review by chadoh — APPROVED** [review]
Looks great, thanks!

**Pull request merged**
